spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -216,9 +220,19 @@ public Object execute(Object[] parameters) {
 							"Too few parameters are provided for query method: " + getQueryMethod().getName());
 				}
 				Object value = it.next();
-				filter.getFieldFilterBuilder().setField(fieldReference)
-						.setOp(getOperator(part, value))
-						.setValue(this.classMapper.toFirestoreValue(value));
+				if (fieldReference.getFieldPath().equals(FieldPath.documentId().toString())) {
+					FirestoreTemplate firestoreTemplate = (FirestoreTemplate) this.reactiveOperations;
+					DocumentReference documentReference = firestoreTemplate.getDocumentReference(persistentEntity.collectionName(), (String) value);
```

Comment:
This field will be the same as the part we are working with in this loop iteration. So you can use `value` instead of `documentReference` when looking up the operation a couple of lines down.